modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -168,7 +175,25 @@ private CompletableFuture<ClusterNode> leaderFuture() {
      */
     public CompletableFuture<? extends NetworkMessage> processPlacementDriverMessage(PlacementDriverReplicaMessage msg) {
         if (msg instanceof LeaseGrantedMessage) {
-            return processLeaseGrantedMessage((LeaseGrantedMessage) msg);
+            return processLeaseGrantedMessage((LeaseGrantedMessage) msg)
+                    .handle((v, e) -> {
```

Comment:
It is sent in the case of not NodeStoppingException.
Actually it fixes another negotiation issue: in the case of any replica exception the agreement is discarded and starts from the beginning on the PD active actor side.

Reply:
That doesn't feel correct. I agree that we should start the negotiation from the beginning in case of exceptions. However, from my point of view, that means that we should have generic exceptional leaseGrantResponse in case of any exception. What's the point of dedicated handling of NodeStopping one?

Reply:
I removed the check for NodeStoppingException.